pkg/gadgets/run/types/metadata.go

```diff
@@ -155,3 +159,134 @@ func (m *GadgetMetadata) validateStructs(spec *ebpf.CollectionSpec) error {
 
 	return result
 }
+
+// Populate fills the metadata  from its ebpf spec
+func (m *GadgetMetadata) Populate(spec *ebpf.CollectionSpec) error {
+	if m.Tracers == nil {
+		m.Tracers = make(map[string]Tracer)
+	}
+
+	if m.Structs == nil {
+		m.Structs = make(map[string]Struct)
+	}
+
+	if err := m.populateTracers(spec); err != nil {
+		return fmt.Errorf("handling trace maps: %w", err)
+	}
+
+	return nil
+}
+
+func (m *GadgetMetadata) populateTracers(spec *ebpf.CollectionSpec) error {
+	traceMap := getTracerMapFromeBPF(spec)
+	if traceMap == nil {
+		log.Debug("No trace map found")
+		return nil
+	}
+
+	traceMapStruct, ok := traceMap.Value.(*btf.Struct)
+	if !ok {
+		return fmt.Errorf("BPF map %q does not have BTF info for values", traceMap.Name)
```

Comment:
```suggestion
		return fmt.Errorf("BPF map %q does not have a BTF struct for its value", traceMap.Name)
```

Reply:
Good point.  I changed the logic to test two different cases: (1) btf info for value missing (2) map value not being a struct.